src/components/NodeMovingPlaceholder/index.test.js

```diff
@@ -0,0 +1,155 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import NodeMovingPlaceholder from './index';
+
+const svg = document.createElement('svg');
+
+const defaultProps = {
+  node: {
+    id: 'node-id',
+    states: [],
+    parents: [],
+    position: { x: 0, y: 0 },
+    size: { width: 160, height: 100 },
+  },
+  svg,
+  onSetPosition: jest.fn(),
+  onCancel: jest.fn(),
+};
+
+const shallowComponent = (props = {}) => {
+  const compProps = { ...defaultProps, ...props };
+
+  return shallow(<NodeMovingPlaceholder {...compProps} />);
+};
+
+describe('NodeMovingPlaceholder Component', () => {
+  describe('Event Listeners', () => {
+    describe('Check bind and unbind events', () => {
+      beforeEach(() => {
+        svg.addEventListener = jest.fn();
+        svg.removeEventListener = jest.fn();
+      });
+
+      it('"mouseup" and "mouseleave" events', () => {
+        const component = shallowComponent();
+        const componentInstance = component.instance();
+
+        expect(svg.addEventListener.mock.calls).toEqual([
+          ['mouseup', componentInstance.handleSetPosition],
+          ['mouseleave', componentInstance.handleCancel],
+        ]);
+
+        return Promise.resolve().then(() => {
+          component.unmount();
+          expect(svg.removeEventListener.mock.calls).toEqual([
+            ['mouseup', componentInstance.handleSetPosition],
+            ['mouseleave', componentInstance.handleCancel],
+          ]);
+        });
+      });
+    });
+
+    describe('When "mouseleave" event is triggered', () => {
+      const map = {};
+      const onCancel = jest.fn();
+
+      beforeAll(() => {
+        svg.addEventListener = jest.fn((event, cb) => {
+          map[event] = cb;
+        });
+      });
+
+      it('calls "onCancel" prop', () => {
+        shallowComponent({ onCancel });
+        map.mouseleave();
+
+        expect(onCancel).toBeCalled();
+      });
+    });
+
+    describe('When "mouseup" event is triggered', () => {
+      let onSetPosition;
+      let map;
+
+      beforeEach(() => {
+        map = {};
+        onSetPosition = jest.fn();
+        svg.addEventListener = jest.fn((event, cb) => {
+          map[event] = cb;
+        });
+      });
+
+      describe('When there is "lastNodePosition" value', () => {
+        it('calls "onSetPosition" prop', () => {
+          const component = shallowComponent({ onSetPosition });
+          component.instance().lastNodePosition = { x: 1, y: 1 };
+          map.mouseup();
+
+          expect(onSetPosition).toBeCalled();
+        });
+      });
+
+      describe('When there is no "lastNodePosition" value', () => {
+        it('calls "onSetPosition" prop', () => {
+          shallowComponent({ onSetPosition });
+          map.mouseup();
+
+          expect(onSetPosition).not.toBeCalled();
+        });
+      });
+    });
+  });
+
+  describe('Methods', () => {
```

Comment:
It is not a good practice to test component methods directly like this, they all should be considered private and you should just interact with the component as a user would. Is there any way it can be tested using simulated events?